cmd/validator/core/validator.go

```diff
@@ -119,7 +123,95 @@ func Start(work <-chan string, processorID ProcessorID, threads int, config Para
 }
 
 // CallProcessor invokes the processor with a retry mechanism.
-func CallProcessor(processor Processor, addrInput string, config Params, results chan<- Result) {
+func CallProcessor(processor Processor, input string, config Params, results chan<- Result) {
+	split := strings.Split(input, ",")
+	if len(split) == 1 {
+		CallProcessorAddress(processor, split[0], config, results)
+	} else {
+		if split[0] == "address" && len(split) == 2 {
+			CallProcessorAddress(processor, split[1], config, results)
+		} else if split[0] == "box" && len(split) == 3 {
+			CallProcessorBox(processor, split[1], split[2], config, results)
+		} else {
+			err := fmt.Errorf("invalid resource")
+			results <- resultResourceError(err, input)
+			return
+		}
+
+	}
+}
+
+func CallProcessorBox(processor Processor, appid string, boxname string, config Params, results chan<- Result) {
+	boxname = strings.ReplaceAll(boxname, "+", "%2B")
+	boxname = strings.ReplaceAll(boxname, "/", "%2F")
+
+	algodDataURL := fmt.Sprintf("%s/v2/applications/%s/box?name=b64:%s", config.AlgodURL, appid, boxname)
+	indexerDataURL := fmt.Sprintf("%s/v2/applications/%s/box?name=b64:%s", config.IndexerURL, appid, boxname)
+
+	// Fetch algod box outside the retry loop. When the data desynchronizes we'll keep fetching indexer data until it
+	// catches up with the first algod box query.
+	algodData, err := getData(algodDataURL, config.AlgodToken)
+	if err != nil {
+		err = fmt.Errorf("error getting algod data (%s): %w", algodData, err)
+		switch {
+		case strings.Contains(string(algodData), api.ErrResultLimitReached):
+			results <- resultBoxSkip(err, appid, boxname, SkipLimitReached)
+		default:
+			results <- resultBoxError(err, appid, boxname)
+		}
+		return
+	}
+
+	// Retry loop.
+	for i := 0; true; i++ {
+		indexerData, err := getData(indexerDataURL, config.IndexerToken)
+		if err != nil {
+			err = fmt.Errorf("error getting indexer data (%s): %w", indexerData, err)
+			switch {
+			case strings.Contains(string(indexerData), api.ErrResultLimitReached):
+				results <- resultBoxSkip(err, appid, boxname, SkipLimitReached)
+			case strings.Contains(string(indexerData), api.ErrNoAccountsFound):
```

Comment:
maybe different errors come out of the applications indexer endpoint that would tell us about a box name or app id not found.